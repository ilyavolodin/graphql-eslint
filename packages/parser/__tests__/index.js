const parser = require('../src');

describe('parser', () => {
    it('parseForESLint() should return ast and tokens', () => {
        const code = `""" generic query placeholder """
            type Query`;

        const result = parser.parseForESLint(code, { fileName: 'test.graphql' });
        expect(result.ast).toMatchSnapshot();
        expect(result.ast.tokens).toBeTruthy();
    });

    it('should throw on invalid code', () => {
        const code = 'Hello World!';

        expect(() => { parser.parseForESLint(code); }).toThrow();
    });

    it('should correctly preserve "type" property', () => {
        const code = `query GetUser($userId: ID!) {
            user(id: $userId) {
                id
            }
        }`;

        const result = parser.parseForESLint(code, { fileName: 'test.graphql' });

        const field = result.ast.definitions[0].variableDefinitions[0];

        expect(field.type).toBe('VariableDefinition');
        expect(field.fieldType.type).toBe('NonNullType');
    });
});
