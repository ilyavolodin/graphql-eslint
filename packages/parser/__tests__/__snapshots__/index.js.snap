// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser parseForESLint() should return ast and tokens 1`] = `
Object {
  "comments": Array [],
  "definitions": Array [
    Object {
      "description": Object {
        "block": true,
        "kind": "StringValue",
        "loc": Object {
          "end": Object {
            "column": 33,
            "line": 1,
          },
          "start": Object {
            "column": 0,
            "line": 1,
          },
        },
        "oldLoc": Object {
          "end": 33,
          "start": 0,
        },
        "range": Array [
          0,
          33,
        ],
        "type": "StringValue",
        "value": " generic query placeholder ",
      },
      "directives": Array [],
      "fields": Array [],
      "interfaces": Array [],
      "kind": "ObjectTypeDefinition",
      "loc": Object {
        "end": Object {
          "column": 22,
          "line": 2,
        },
        "start": Object {
          "column": 0,
          "line": 1,
        },
      },
      "name": Object {
        "kind": "Name",
        "loc": Object {
          "end": Object {
            "column": 22,
            "line": 2,
          },
          "start": Object {
            "column": 17,
            "line": 2,
          },
        },
        "oldLoc": Object {
          "end": 56,
          "start": 51,
        },
        "range": Array [
          51,
          56,
        ],
        "type": "Name",
        "value": "Query",
      },
      "oldLoc": Object {
        "end": 56,
        "start": 0,
      },
      "range": Array [
        0,
        56,
      ],
      "type": "ObjectTypeDefinition",
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": Object {
      "column": 22,
      "line": 2,
    },
    "start": Object {
      "column": 0,
      "line": 1,
    },
  },
  "oldLoc": Object {
    "end": 56,
    "start": 0,
  },
  "range": Object {},
  "tokens": Array [
    Object {
      "column": 1,
      "end": 33,
      "kind": "BlockString",
      "line": 1,
      "next": null,
      "prev": Object {
        "column": 0,
        "kind": "<SOF>",
        "line": 0,
        "value": undefined,
      },
      "range": Array [
        0,
        33,
      ],
      "start": 0,
      "value": " generic query placeholder ",
    },
    Object {
      "column": 13,
      "end": 50,
      "kind": "Name",
      "line": 2,
      "next": null,
      "prev": Object {
        "column": 1,
        "kind": "BlockString",
        "line": 1,
        "value": " generic query placeholder ",
      },
      "range": Array [
        46,
        50,
      ],
      "start": 46,
      "value": "type",
    },
    Object {
      "column": 18,
      "end": 56,
      "kind": "Name",
      "line": 2,
      "next": null,
      "prev": Object {
        "column": 13,
        "kind": "Name",
        "line": 2,
        "value": "type",
      },
      "range": Array [
        51,
        56,
      ],
      "start": 51,
      "value": "Query",
    },
  ],
  "type": "Document",
}
`;
